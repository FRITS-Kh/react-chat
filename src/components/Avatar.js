import React from 'react';
import PropTypes from 'prop-types';
import MUIAvatar from '@material-ui/core/Avatar';
import titleInitials from '../utils/title-initials';
import getColor from '../utils/color-from';

const Avatar = ({ name, colorFrom, ...rest }) => (
  <MUIAvatar style={{ backgroundColor: getColor(colorFrom || name) }} {...rest}>
    {titleInitials(name)}
  </MUIAvatar>
);

Avatar.propTypes = {
  name: PropTypes.string.isRequired,
  colorFrom: PropTypes.string.isRequired,
};

export default Avatar;
