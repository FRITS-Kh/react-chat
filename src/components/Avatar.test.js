/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Avatar from './Avatar';

describe('<Avatar />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Avatar colorFrom="12345" name="Name Surname" />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Avatar colorFrom="12345" name="Name Surname" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
