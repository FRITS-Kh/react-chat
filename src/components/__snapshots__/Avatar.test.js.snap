// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> renders correctly 1`] = `
<div
  className="MuiAvatar-root-4 MuiAvatar-colorDefault-5"
  style={
    Object {
      "backgroundColor": "#ffc107",
    }
  }
>
  NS
</div>
`;
